sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/SchemaUtils.java

```diff
@@ -128,18 +128,15 @@ private static ColumnSchema fromBeamField(Schema.Field field) {
       }
       ColumnSchema column =
           fromBeamField(Field.of(field.getName(), fieldType.getCollectionElementType()));
-      if (!column.getMode().isEmpty()) {
```

Comment:
This assertion could still be there and change to `column.getMode().equals("REQUIRED")`

Reply:
Done.

I think this check is redundant for now because the only place that sets mode is the else branch below that the array element check will surely enter, because we already checked array element is not ARRAY type. And we have also checked the array element is not nullable above so it is always `REQUIRED`. But I agree leaving this check here can prevent code change that breaks this in the future.